tests/integration/test_ls240_agent_integration.py

```diff
@@ -7,112 +7,81 @@
+    client.main.stop()
+    resp = client.main.wait()
     print(resp)
-    assert resp.status == ocs.OK
-    print(resp.session)
+    print(resp.session['data'])
     assert resp.session['op_code'] == OpCode.SUCCEEDED.value
+    assert resp.session['data']['fields']['Channel_1']['V'] == 1000
 
 
 @pytest.mark.integtest
-def test_ls240_start_acq(wait_for_crossbar, emulator, run_agent, client):
-    client.init_lakeshore()
-
-    responses = {'*IDN?': 'LSCI,MODEL240,LSA240S,1.3',
-                 'KRDG? 1': '+1.0E-03',
-                 'SRDG? 1': '+1.0E+03',
-                 'KRDG? 2': '+1.0E-03',
-                 'SRDG? 2': '+1.0E+03',
-                 'KRDG? 3': '+1.0E-03',
-                 'SRDG? 3': '+1.0E+03',
-                 'KRDG? 4': '+1.0E-03',
-                 'SRDG? 4': '+1.0E+03',
-                 'KRDG? 5': '+1.0E-03',
-                 'SRDG? 5': '+1.0E+03',
-                 'KRDG? 6': '+1.0E-03',
-                 'SRDG? 6': '+1.0E+03',
-                 'KRDG? 7': '+1.0E-03',
-                 'SRDG? 7': '+1.0E+03',
-                 'KRDG? 8': '+1.0E-03',
-                 'SRDG? 8': '+1.0E+03'}
-    emulator.define_responses(responses)
-
-    resp = client.acq.start(sampling_frequency=1.0)
-    assert resp.status == ocs.OK
-    assert resp.session['op_code'] == OpCode.STARTING.value
-
-    # We stopped the process with run_once=True, but that will leave us in the
-    # RUNNING state
-    resp = client.acq.status()
-    assert resp.session['op_code'] == OpCode.RUNNING.value
-
-    # Now we request a formal stop, which should put us in STOPPING
-    client.acq.stop()
-    # this is so we get through the acq loop and actually get a stop command in
-    # TODO: get sleep_time in the acq process to be small for testing
-    time.sleep(3)
-    resp = client.acq.status()
-    print(resp)
-    print(resp.session)
-    assert resp.session['op_code'] in [OpCode.STOPPING.value,
-                                       OpCode.SUCCEEDED.value]
-
-
-@pytest.mark.integtest
-def test_ls240_set_values(wait_for_crossbar, emulator, run_agent, client):
-    client.init_lakeshore()
-
+def test_ls240_set_values(wait_for_crossbar, emulator: DeviceEmulator, run_agent, client) -> None:
     responses = {'INNAME 1,Channel 01': '',
                  'INTYPE 1,1,1,0,0,1,1': ''}
-    emulator.define_responses(responses)
+    emulator.update_responses(responses)
 
-    resp = client.set_values(channel=1,
-                             sensor=1,
-                             auto_range=1,
-                             range=0,
-                             current_reversal=0,
-                             units=1,
-                             enabled=1,
-                             name="Channel 01")
+    set_values_params = LS240Actions.SetValues(
+        channel=1, sensor=1, auto_range=1, range=0, current_reversal=0, units=1,
+        enabled=1, name="Channel 01"
+    )
+    resp = client.set_values(**asdict(set_values_params))
```

Comment:
Clients don't need to work this way, and using regular keyword args still works, however I think it offers a couple of benefits... Mainly it ensures that client keyword args are what the agent expects, and it allows for static type checking on the client-side. If there is an unexpected parameter, a missing required parameter, or a type mismatch it can be caught here statically as opposed to requiring an integration test.